sdk/core/src/error/http_error.rs

```diff
@@ -89,46 +90,87 @@ struct ErrorDetails {
 }
 
 impl ErrorDetails {
-    fn new(headers: &HashMap<String, String>, body: &[u8]) -> Self {
-        let mut code = get_error_code_from_header(headers);
-        code = code.or_else(|| get_error_code_from_body(body));
-        let message = get_error_message_from_body(body);
-        Self { code, message }
+    fn new(headers: &Headers, body: &[u8]) -> Self {
+        let header_err_code = get_error_code_from_header(headers);
+        let content_type = headers.get_optional_str(&headers::CONTENT_TYPE);
+        let (body_err_code, body_err_message) =
+            get_error_code_message_from_body(body, content_type);
+
+        let code = header_err_code.or(body_err_code);
+        Self {
+            code,
+            message: body_err_message,
+        }
     }
 }
 
 /// Gets the error code if it's present in the headers
 ///
 /// For more info, see [here](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#handling-errors)
-fn get_error_code_from_header(headers: &HashMap<String, String>) -> Option<String> {
-    headers.get(headers::ERROR_CODE.as_str()).cloned()
+pub(crate) fn get_error_code_from_header(headers: &Headers) -> Option<String> {
+    headers.get_optional_string(&headers::ERROR_CODE)
 }
 
 #[derive(Deserialize)]
 struct NestedError {
+    #[serde(alias = "Message")]
     message: Option<String>,
+    #[serde(alias = "Code")]
     code: Option<String>,
```

Comment:
[REST API guidelines](https://aka.ms/azapi/guidelines) actually recommend camelCase. If you're seeing these PascalCase, that's an anomaly that shouldn't be accounted for here. None of our other languages even deserialize error details and this will eventually be removed anyway. If there's a service that has details you need, I suggest deserialize your own models or - for now, anyway - you limit it to the crate that is different i.e., the one using PascalCase.

If that's only for XML, this would hamper JSON deserialization as well. Some other way may be necessary even to the point, if possible (though I don't think it is for stock serde) to ignore case.

Reply:
It is only for XML which is not included into the REST API guidelines and for XML PascalCase is the norm.
Would you prefer to have a separate `struct XmlErrorBody`? I did it using `serde(alias =)` because it required the least amount of code and thought the internal serde overhead would be negligible

Reply:
No, this is fine. My apologies. For some reason I was thinking of `rename` instead of `alias`. I think this is fine.